onnx/defs/data_propagators.h

```diff
@@ -16,4 +49,160 @@ inline void PropagateShapeDataFromInputToOutput(DataPropagationContext& ctx, int
   }
 }
 
+// Data propagation function for Shape op
+// Propagates input shape to output shape
+inline void ShapeOpDataPropagator(DataPropagationContext& ctx) {
+  if (!hasNInputShapes(ctx, 1)) {
+    return;
+  }
+  if (ctx.getInputType(0)->tensor_type().has_shape()) {
+    auto input_shape = ctx.getInputType(0)->tensor_type().shape();
+    TensorShapeProto tsp;
+    tsp.CopyFrom(input_shape);
+    ctx.addOutputData(0, std::move(tsp));
+  }
+}
+
+inline void SizeOpDataPropagator(DataPropagationContext& ctx) {
+  const auto input_data = ctx.getInputData(0);
+  if (input_data != nullptr) {
+    TensorShapeProto tsp;
+    tsp.mutable_dim()->Add()->set_dim_value(input_data->dim_size());
```

Comment:
this is not correct ... size returns total number of elements... you should iterator over the dims and calculate the number of elements

Reply:
After offline discussion, the logic here should be good. I have updated Size unit test to prevent confusion.